lib/ddtrace/logger.rb

```diff
@@ -10,30 +11,89 @@ class Logger < ::Logger
     def initialize(*args, &block)
       super
       self.progname = LOG_PREFIX
+
+      # Rate limit buckets
+      @buckets = {}
     end
 
     def add(severity, message = nil, progname = nil, &block)
```

Comment:
There's quite a bit of new logic added here. In case something goes wrong and it throws an error, are we handling that such that it won't crash or break the user application?